test/controller_test.py

```diff
@@ -0,0 +1,214 @@
+import unittest
+from unittest import runner
+import pygame
+import pygame._sdl2.controller as controller
+from pygame.tests.test_utils import prompt, question
+
+class ControllerModuleTest(unittest.TestCase):
+    def setUp(self):
+        controller.init()
+
+    def tearDown(self):
+        controller.quit()
+
+    def test_init(self):
+        controller.quit()
+        controller.init()
+        self.assertTrue(controller.get_init())
+
+    def test_init__multiple(self):
+        controller.init()
+        controller.init()
+        self.assertTrue(controller.get_init())
+
+    def test_quit(self):
+        controller.quit()
+        self.assertFalse(controller.get_init())
+
+    def test_quit__multiple(self):
+        controller.quit()
+        controller.quit()
+        self.assertFalse(controller.get_init())
+
+    def test_get_init(self):
+        self.assertTrue(controller.get_init())
+
+    def test_set_eventstate(self):
```

Comment:
I wonder if this needs to be an interactive test? Or maybe split into a part which requires interaction.

"Please move the controller or press some buttons on it"